src/Patient_Dashboard/PatientDashboard.js

```diff
@@ -1,60 +1,203 @@
+          ? [...prevPreviews[visitId], previewUrl]
+          : [previewUrl],
+      }));
+      // Pass the visit object to the UploadDocumentPage
+      const selectedVisit = visits.find((visit) => visit.uuid === visitId);
+      // const visitDate = formatDate(selectedVisit.display); // Extract visitDate
+      navigate(`/upload/${visitId}`, {
+        state: {
+          patientData,
+          visit: selectedVisit, // Pass the visit object here
+          selectedDocumentPreview: previewUrl,
+          visitDate: visitDate, // Pass the visitDate here
+        },
+      });
+    }
+  };
   const getInitials = (name) => {
-    if (typeof name === 'string') {
+    if (typeof name === "string") {
       const initials = name
-        .split(' ')
+        .split(" ")
         .map((word) => word.charAt(0).toUpperCase())
-        .join('');
+        .join("");
       return initials;
     }
-    return '';
+    return "";
   };
-
   return (
     <DashboardLayout>
-      <div className="patient-search-container">
-        {patientData ? (
+      <div className="dashboard-container">
+        <div className="patient-search-container">
           <div className="patient-card-container">
             <div className="patient-photo">
-              {patientData.photoUrl ? (
+              {patientData && patientData.photoUrl ? (
                 <img src={patientData.photoUrl} alt="Patient" />
               ) : (
-                <div className="initials-icon">{getInitials(patientData.name)}</div>
+                <div className="initials-icon">
+                  {patientData && getInitials(patientData.name)}
+                </div>
               )}
             </div>
             <div className="patient-info">
               <div className="patient-details">
-                <h2>{patientData.name}</h2>
-                <div className="patient-id">ID: {patientData.identifier}</div>
+                <h2>{patientData && patientData.name}</h2>
+                <div className="patient-id">
+                  {patientData && `ID: ${patientData.identifier}`}
+                </div>
               </div>
             </div>
           </div>
-        ) : (
-          <div>Patient data not found.</div>
-        )}
-        <div className="search-box-container">
-          <Search
-            size="lg"
-            labelText="Search"
-            placeholder="Search"
-            closeButtonLabelText="Clear search input"
-            id="search-1"
-            onChange={() => {}}
-            onKeyDown={() => {}}
-          />
+          <div className="search-box-container">
+            <Search
+              size="lg"
+              labelText="Search"
+              placeholder="Search"
+              closeButtonLabelText="Clear search input"
+              id="search-1"
+              onChange={() => {}}
+              onKeyDown={() => {}}
+            />
+          </div>
+        </div>
+        <div className="visits-container">
```

Comment:
The "view code" and "Domain Logic" is mixed in this class. Patient Name/Initials/etc.. can come from separate JS code, and React Code should only have display logic?